spec/models/partner_spec.rb

```diff
@@ -174,6 +174,15 @@
     end
   end
 
+  describe '#contact_person' do
+    let(:partner) { create(:partner) }
+
+    it "checks for agency in response before fetching contact info" do
+      allow(partner).to receive(:partnerbase_partner) { OpenStruct.new(agency: nil) }
```

Comment:
```suggestion
      allow(partner).to receive(:partnerbase_partner) { instance_double('partnerbase_partner', agency: nil) }
```

I feel like using `instance_double` here is a bit more alike to what is used in tests. Let me know if this works.

Reply:
This update has been made.